src/ucs/sys/sys.c

```diff
@@ -1158,3 +1179,39 @@ void ucs_sys_cpuset_copy(ucs_cpu_set_t *dst, const ucs_sys_cpuset_t *src)
         }
     }
 }
+
+ucs_sys_ns_t ucs_sys_get_ns(ucs_sys_ns_name_t name)
+{
+    static ucs_init_once_t init_once = UCS_INIT_ONCE_INITIALIZER;
+    char filename[MAXPATHLEN];
+    int res;
+    struct stat st;
+    ucs_sys_ns_name_t ns;
+
+    if (name >= UCS_SYS_NS_LAST) {
+        return 0;
+    }
+
+    UCS_INIT_ONCE(&init_once) {
+        for (ns = UCS_SYS_NS_IPC; ns < UCS_SYS_NS_LAST; ns++) {
+            snprintf(filename, sizeof(filename), "%s/%s", UCS_PROCESS_NS_DIR,
+                     ucs_sys_namespace_info[ns].name);
+
+            res = stat(filename, &st);
+            if (res == 0) {
+                ucs_sys_namespace_info[ns].ino = (ucs_sys_ns_t)st.st_ino;
+            } else {
+                ucs_sys_namespace_info[ns].ino = ucs_sys_namespace_info[ns].dflt;
```

Comment:
maybe also print a warning with errno?

Reply:
not sure if freeBSD or MacOS support such bood ID - it could be ok if such file is missing